### ISSUE TITLE
Support SSL connections to PostgreSQL

### ISSUE THREAD

**Issue by billux** [maplibre/martin#63]
As mentionned [here](https://github.com/urbica/martin/issues/17#issuecomment-536903060), martin doesn't support SSL connections to PostgreSQL since `TlsMode` is hardcoded to `None` [here](https://github.com/urbica/martin/blob/master/src/db.rs#L12).

Is it possible to make this configurable, either using `sslmode` variable in the connection string or with a specific setting?

Thank!

**Comment by lkleuver** [comment]
Is this being added in the very near future? 
We use a digitalocean managed DB which requires SSL.
Would really like keeping Martin but otherwise, we will need to find a different tile server solution.

I'm still in the very early stages of learning Rust so I'm afraid I can't really contribute (yet)

**Comment by StefanBrand** [comment]
@lkleuver It seems it has been merged to master (#90), so you should be able to build martin yourself.

**Comment by douglasg14b** [comment]
@stepankuzmin Will there be an updated release that contains these changes?

**Comment by shaunakv1** [comment]
I second, that, this is a critically needed feature @stepankuzmin !

**Comment by stepankuzmin** [comment]
Hi everyone,

Sorry for the extended response, and thanks for your patience. I've published the pre-release version [v1.0.0-alpha.0](https://github.com/urbica/martin/releases/tag/v1.0.0-alpha.0), and updated [README](https://github.com/urbica/martin#recipes) adding two recipes of using Martin with DigitalOcean and Heroku Managed PostgreSQL.

Feel free to try it as a `urbica/martin:1.0.0-alpha.0` docker image.

### Using with DigitalOcean PostgreSQL

You can use Martin with [Managed PostgreSQL from DigitalOcean](https://www.digitalocean.com/products/managed-databases-postgresql/) with PostGIS extension

First, you need to download the CA certificate and get your cluster connection string from the [dashboard](https://cloud.digitalocean.com/databases). After that, you can use the connection string and the CA certificate to connect to the database.

```
martin --ca-root-file ./ca-certificate.crt postgres://user:password@host:port/db?sslmode=require
```

### Using with Heroku PostgreSQL

You can use Martin with [Managed PostgreSQL from Heroku](https://www.heroku.com/postgres) with PostGIS extension

```
heroku pg:psql -a APP_NAME -c 'create extension postgis'
```

To trust the Heroku certificate, you can disable certificate validation with either the `DANGER_ACCEPT_INVALID_CERTS` environment variable

```
DATABASE_URL=$(heroku config:get DATABASE_URL -a APP_NAME) DANGER_ACCEPT_INVALID_CERTS=true martin
```

or `--danger-accept-invalid-certs` command-line argument

```
martin --danger-accept-invalid-certs $(heroku config:get DATABASE_URL -a APP_NAME)
```

**Comment by shaunakv1** [comment]
@stepankuzmin  Thanks for this fix! Just tested this on azure webapp backed by azure postgresql  and it works well.  We run our webapp and postgres on a subnet and trust the certificate so I just set the `DANGER_ACCEPT_INVALID_CERTS=true` , in case it helps anyone.
